File: Palindrome.cpp
bool isPalindrome(long x){

long reversed_number=0;
int a=x;
if(x<0)
{
    return false;
    return 0;
}

 if(x>=0 && x<=9)
{
    return true;
    return 0;
}
if(x%10==0)
{
    return false;
}

if(x>=-2^31||x<=2^31-1)
{

    while(x>0)
    {
        reversed_number=reversed_number*10 + x%10;
        x/=10;
    }
}


if(a==reversed_number)
{
    return true;
}
return false;
}

File: Palindrome.java
class Main {
  public static void main(String[] args) {
    
    int number = 7865, reversednumber = 0, remainder;
    
    int originalnumber = number;
    
    while (number != 0) {
      remainder = number % 10;
      reversednumber = reversednumber * 10 + remainder;
      number /= 10;
    }
    
    if (originalnumber == reversednumber) {
      System.out.println(originalnumber + " is Palindrome number.");
    }
    else {
      System.out.println(originalnumber + " is not Palindrome number.");
    }
  }
}
